services/movie_session.py

```diff
@@ -0,0 +1,56 @@
+from typing import List, Optional
+from datetime import datetime
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(
+        session_date: Optional[str] = None
+) -> List[MovieSession]:
+    movie_sessions = MovieSession.objects.all()
+
+    if session_date:
+        movie_sessions = movie_sessions.filter(show_time__date=session_date)
+
+    return movie_sessions
+
+
+def get_movie_session_by_id(session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: Optional[datetime] = None,
+        movie_id: Optional[int] = None,
+        cinema_hall_id: Optional[int] = None
+) -> None:
+    movie_session = MovieSession.objects.get(id=session_id)
+
+    if show_time:
+        movie_session.show_time = show_time
+    if movie_id:
+        movie_session.movie_id = movie_id
+    if cinema_hall_id:
+        movie_session.cinema_hall_id = cinema_hall_id
+
+    movie_session.save()
+
+
+def delete_movie_session_by_id(session_id: int) -> None:
+    movie_session = get_movie_session_by_id(session_id)
+
+    if movie_session:
+        movie_session.delete()
```

Comment:
You could implement the logic of this method in one line without allocating any additional variables. Take into account that the `if` check is not needed since the error would be raised earlier in the `get_movie_session_by_id` method if record with this `id` would be absent.